mitodl/open-discussions
# Various fixes for ETL loading bugs

#### Pre-Flight checklist

#### What are the relevant tickets?
Fixes #2961 - was causing video topics to not load completely
Part of #2955 

#### What's this PR do?
- Removes `course_catalog.etl.utils.log_exceptions` - this was a minor ergonomic improvement to writing code but a huge ergonomic detriment to testing it
- Fixes the course deduplication code to query on `platform` since otherwise it may select a course from a different platform. In practice we saw this happen between OCW + MITx courses
- Fixes a race condition (#2961) in writing out topic data. I solved this by adding transactions and taking locks on the parent resource.

#### How should this be manually tested?

- Verify you're seeing a course that doesn't appear in the API response. Message me for some examples.
- Run `./manage.py backpopulate_xpro_data` and ./manage.py backpopulate_edx_data`, they should succeed and you should see the course get unpublished.
- Adds a concept of loader configurations, to allow per-platform configuration of loaders, specifically:
  - Some integrations need to be able to treat `offered_by` as an appendable relation
  - Some integrations don't remove courses (SEE, CSAIL, MITPE) because we are parsing html pages which are sometimes updated in such ways that we can't parse a current course, but we don't want to delist existing courses
